src/modules/processing/services/instagram.js

```diff
@@ -196,21 +239,29 @@ function extractNewPost(data, id) {
 }
 
 async function getPost(id) {
-    let data, result, dataType = 'old';
+    let data, result, dataType = 'new';
     try {
         const cookie = getCookie('instagram');
 
-        data = await requestHTML(id);
+        // mobile api (no cookie, cookie)
+        data = await requestMobileApi(id);
+        if (!data && cookie) data = await requestMobileApi(id, cookie);
+
+        // html embed (no cookie, cookie)
+        if (!data) data = await requestHTML(id);
         if (!data && cookie) data = await requestHTML(id, cookie);
 
-        if (!data) {
-            dataType = 'new';
-            data = await requestGQL(id, cookie);
-        }
+        // web app graphql api (no cookie, cookie)
+        if (!data) data = await requestGQL(id);
+        if (!data && cookie) data = await requestGQL(id, cookie);
     } catch {}
 
     if (!data) return { error: 'ErrorCouldntFetch' };
 
+    if (data?.gql_data) {
+        dataType = 'old'
+    }
+
     if (dataType === 'new') {
```

Comment:
maybe just use `data?.gql_data` as condition directly and rename functions to `extractAPIPost` and `extractGQLPost` or something

Reply:
![image](https://github.com/wukko/cobalt/assets/71202418/628a1038-5d17-48c1-a5d1-c22eca763e09)